test/libsolidity/semanticTests/inlineAssembly/keccak256_optimizer_bug_different_memory_location.sol

```diff
@@ -0,0 +1,16 @@
+contract C {
+    function f() public view returns (bool ret) {
+        assembly {
+            let x := calldataload(0)
+            mstore(0, x)
+            mstore(0x20, x)
+            let a := keccak256(0, 4)
+            let b := keccak256(0x20, 8)
+            ret := eq(a, b)
+        }
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// f() -> false
```

Comment:
Could you please add a positive example for intended optimisation? Perhaps this

```
contract C {
    function f() public view returns (bool ret) {
        assembly {
            let x := calldataload(0)
            mstore(0, x)
            mstore(0x20, x)
            let a := keccak256(0, 4)
            let b := keccak256(0x20, 4)
            ret := eq(a, b)
        }
    }
}
// ====
// compileViaYul: also
// ----
// f() -> true
```

Reply:
Added this.